crates/solver/src/liquidity/zeroex.rs

```diff
@@ -136,8 +135,8 @@ impl LiquidityCollecting for ZeroExLiquidity {
         pairs: HashSet<TokenPair>,
         _block: Block,
     ) -> Result<Vec<Liquidity>> {
-        let zeroex_orders = self.orderbook_cache.read().await.clone();
-        let order_buckets = generate_order_buckets(zeroex_orders, pairs);
+        let zeroex_orders = self.orderbook_cache.load().clone();
+        let order_buckets = generate_order_buckets((*zeroex_orders).clone(), pairs);
         let filtered_zeroex_orders = get_useful_orders(order_buckets, 5);
```

Comment:
I just noticed now that we are not getting a fresh set of `OrderRecord`s for every request `generate_order_buckets()` and `get_useful_orders() should be rewritten such that they only operate on references and only clone the values that will actually get returned.